### DISCUSSION TITLE
I've made a JS lib

### DISCUSSION THREAD

**Discussion by thurti** [ggml-org/whisper.cpp#2125] [Show and tell]
I've created a JavaScript library based on whisper.cpp. This started as part of another project but then I got sidetracked and had fun to make this a npm package with docs and all.

Repo: https://github.com/TranscribeJs/transcribe.js  
Website: https://www.transcribejs.dev

My goal was to have something short like `await transcribe('myfile.mp4')` to get the full json output. I've made a custom wasm build (copied together from the examples) and a js class that handles the communication with the wasm module and the whole audio and file related stuff.

Maybe someone will find this useful.

**Comment by fcyrizz** [comment]
is there any way to deploy it on vercel, using the "tiny.en-q5_1" model?

**Reply by thurti** [reply]
You propably need to add cross origin [http header](https://www.transcribejs.dev/docs/prerequisite/) for the wasm js files. I've never used vercel myself but I think you can do this with the project [config](https://vercel.com/docs/projects/project-configuration#headers).
